### ISSUE TITLE
Can't import libraries in virtual environment despite installing bpython in that context

### ISSUE THREAD

**Issue by thomasballinger** [bpython/bpython#650]
I tried to run bpython in a virtual environment, but since I hadn't installed it in that virtual environment I couldn't import `numpy`, which I had installed for the virtual environment but not for my system Python.

So I ran `pip install bpython` and tried bpython again, but I still couldn't import that library! What's going on?

**Comment by thomasballinger** [comment]
tl;dr run `hash -r` in bash, `rehash` in zsh; or deactivate and reactivate the virtual environment.

Great question! This is an issue with the way your shell decides which executable file to run when you type a command. The first time during a shell session you type a command like `bpython` and hit enter, each directory in the `$PATH` is searched* for an executable file with that name. Then that file is run, but not before caching that mapping of command to file path. Then the second time you run `bpython`, the executable file run the first time will still be used. 

By typing `hash -r` in bash (`rehash` in some other shells) you can tell bash to forget these saved locations. Or you can deactivate the virtual environment and reactivate it, or open a new terminal.

*I don't actually know the exact search behavior, it's probably more complicated than this, see [man bash](https://linux.die.net/man/1/bash) and search for `hash` for more info.
